lib/Finance/Calendar.pm

```diff
@@ -864,7 +864,9 @@ sub _market_opens {
                 my $int_close = $breaks[$i][1];
                 my $next_open = exists $breaks[$i + 1] ? $breaks[$i + 1][0] : $close;
 
-                if ($when->is_after($current_open) and $when->is_before($int_open)) {
+                if ($when->is_before($int_open)
+                    and ($when->is_same_as($current_open) or $when->is_after($current_open)))
```

Comment:
Actually, why not just change it to:

```
$when->epoch >= $current_open->epoch
```

Reply:
Yeah, that's possible. It's just in the rest of the code we used the same pattern. If you think it adds value I think it's better to do it as a refactoring in another card.